File: razarion-client/src/main/java/com/btxtech/client/cockpit/radar/MiniViewField.java
package com.btxtech.client.cockpit.radar;

import com.btxtech.uiservice.renderer.ViewField;
import elemental.html.CanvasRenderingContext2D;

import javax.enterprise.context.Dependent;

/**
 * Created by Beat
 * on 16.06.2017.
 */
@Dependent
public class MiniViewField extends AbstractGameCoordinates {
    private static final double LINE_WIDTH = 2;

    @Override
    protected void draw(CanvasRenderingContext2D ctx) {
        ViewField viewField = getViewField();

        getCtx().setLineWidth(toCanvasPixel(LINE_WIDTH));

        getCtx().setStrokeStyle("#fff");
        getCtx().beginPath();
        getCtx().moveTo((float) viewField.getBottomLeft().getX(), (float) viewField.getBottomLeft().getY());
        getCtx().lineTo((float) viewField.getBottomRight().getX(), (float) viewField.getBottomRight().getY());
        getCtx().lineTo((float) viewField.getTopRight().getX(), (float) viewField.getTopRight().getY());
        getCtx().lineTo((float) viewField.getTopLeft().getX(), (float) viewField.getTopLeft().getY());
        getCtx().closePath();
        getCtx().stroke();
    }
}

File: razarion-frontend/src/app/game/cockpit/main/radar/mini-view-field.ts
import { GameUiControl } from 'src/app/gwtangular/GwtAngularFacade';
import { AbstractGameCoordinates } from './abstract-game-coordinates';
import { BabylonRenderServiceAccessImpl } from 'src/app/game/renderer/babylon-render-service-access-impl.service';
import {ViewField} from "../../../renderer/view-field";

export class MiniViewField extends AbstractGameCoordinates {
    private static readonly LINE_WIDTH = 1;

    constructor(gameUiControl: GameUiControl, renderService: BabylonRenderServiceAccessImpl) {
        super(gameUiControl, renderService);
    }

    protected draw(ctx: CanvasRenderingContext2D): void {
        let viewField: ViewField = this.getViewField();

        ctx.lineWidth = this.toCanvasPixel(MiniViewField.LINE_WIDTH);

        ctx.strokeStyle = "#fff";
        ctx.beginPath();
        ctx.moveTo(viewField.getBottomLeft().getX(), viewField.getBottomLeft().getY());
        ctx.lineTo(viewField.getBottomRight().getX(), viewField.getBottomRight().getY());
        ctx.lineTo(viewField.getTopRight().getX(), viewField.getTopRight().getY());
        ctx.lineTo(viewField.getTopLeft().getX(), viewField.getTopLeft().getY());
        ctx.closePath();
        ctx.stroke();
    }
}
